### PR TITLE
Bump up minimum Python version to 3.10

### PR DESCRIPTION
**Pull request by algochoi**
This PR deletes the outdated travis configurations, and updates the python_requires version from 3.5 to 3.10. It also updates the Circle/Docker image Python versions to `3.10.x` as well. 

Currently, the minimum Python version supported by pynacl, one of our dependencies, is 3.6. It makes sense for our SDK to move to a higher version as well. Since PyTeal depends on this repo, we should also bump up the minimum Python version there too.

Closes https://github.com/algorand/py-algorand-sdk/issues/300

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ workflows:
 jobs:
   unit-test:
     docker:
-      - image: python:3.7.9
+      - image: python:3.10.4
```
Comment (jasonpaulos) [inline .circleci/config.yml:14]: Why not just use 3.10? I think the patch version need not be pinned down

Comment (algochoi) [reply .circleci/config.yml]: Good point - updated

File: docs/requirements.txt [hunk 2]
```diff
@@ -1,3 +1,3 @@
-sphinx==4.0.1
+sphinx==4.2.0
```
Comment (algochoi) [inline docs/requirements.txt:1]: sphinx 4.2.0 is the first version that seems to work with Python 3.10, see discussion [here](https://github.com/sphinx-doc/sphinx/issues/9562)

**Review by michaeldiamant — APPROVED** [review]
@algochoi Thanks for the effort to bump the minimum Python version.  

Take my approval with a grain of salt due to my lack of py-algorand-sdk experience.  Here's what I checked:
* Visually sanity checked build steps to confirm tests executing as intended.
* Opened a draft PR to confirm pyteal works as a consumer:  https://github.com/algorand/pyteal/pull/275.
* Looked for missed Python version references via ` git grep "3\."`.

**Review by jasonpaulos — APPROVED** [review]
Looks good

**Review by michaeldiamant — CHANGES_REQUESTED** [review]
@algochoi Sorry for the late change, but I dismissed my approval due to https://github.com/algorand/pyteal/pull/276.

Let's wait until https://github.com/algorand/pyteal/pull/276 is merged to confirm our understanding.

It feels like the repo ought to define a `.readthedocs.yaml`.  I don't (yet) have a readthedocs login, so I don't know what web configurations exist.  And it seems preferable to rely on a local, versioned config.

File: docs/requirements.txt [hunk 3]
```diff
@@ -1,3 +1,3 @@
...
 sphinx-rtd-theme
```
Comment (michaeldiamant) [inline docs/requirements.txt:2]: @algochoi With the _suggestion_ below, I think the PR is ready for approval.

https://github.com/algorand/pyteal/pull/279 is the corresponding change set that fixed the PyTeal readthedocs build.  Thanks for reflecting `.readthedocs.yaml` into the PR.

```suggestion
sphinx-rtd-theme==1.0.0
```

Comment (algochoi) [reply docs/requirements.txt]: Thanks for the catch - will commit the suggestion

**Review by jasonpaulos — APPROVED** [review]
Readthedocs changes look good

**Review by michaeldiamant — APPROVED** [review]
@algochoi Thanks for the effort here to bring py-algorand-sdk up to Python 3.10!  ☕

**Pull request merged**